packages/@aws-cdk/cdk/lib/removal-policy.ts

```diff
@@ -1,34 +1,27 @@
-import { CfnResource } from './cfn-resource';
-import { DeletionPolicy } from './resource-policy';
-
 export enum RemovalPolicy {
   /**
-   * This is the default removal policy for most resources. It means that when the resource
-   * is removed from the app, it will be physically destroyed.
+   * This is the default removal policy. It means that when the resource is
+   * removed from the app, it will be physically destroyed.
    */
-  Destroy = 0,
+  Destroy = 'destroy',
 
   /**
    * This uses the 'Retain' DeletionPolicy, which will cause the resource to be retained
    * in the account, but orphaned from the stack.
    */
-  Orphan,
+  Retain = 'retain',
+}
 
+export interface RemovalPolicyOptions {
   /**
-   * This will apply the 'Retain' DeletionPolicy and also add metadata for the toolkit
-   * to apply a CloudFormation stack policy which forbids the deletion of resource.
+   * The default policy to apply in case the removal policy is not defined.
+   *
+   * @default RemovalPolicy.Retain
    */
-  Forbid
-}
-
-export function applyRemovalPolicy(resource: CfnResource, removalPolicy: RemovalPolicy | undefined) {
-  if (removalPolicy === RemovalPolicy.Orphan || removalPolicy === RemovalPolicy.Forbid) {
-    resource.options.deletionPolicy = DeletionPolicy.Retain;
-  }
+  readonly default?: RemovalPolicy;
 
-  // attach metadata that will tell the toolkit to protect this resource by
-  // applying an appropriate stack update policy.
-  if (removalPolicy === RemovalPolicy.Forbid) {
-    resource.node.addMetadata('aws:cdk:protected', true);
-  }
-}
+  /**
+   * Apply the same deletion policy to the resource's "UpdateReplacePolicy"
+   */
+  readonly applyToUpdateReplacePolicy?: boolean;
```

Comment:
I suppose it's good that you're doing a bit-for-bit replacement of functionality, but I actually think it's a bug if this is ever set to `false`. We should always treat updates the same as deletes (because they will be physical deletes).

Reply:
I was just unsure whether applying this everywhere will break something and felt it's not the right thing to spend my time on.

Reply:
Alright then: https://github.com/awslabs/aws-cdk/issues/2901